web/test.html

```diff
@@ -1,110 +1,210 @@
+    writeToScreen('error: ' + evt.data + '\n');
+    websocket.close();
+    document.myform.connectButton.disabled = false;
+    document.myform.disconnectButton.disabled = true;
+
+  }
+
+  //Function called when send request to Web-Socket
+  function doSend(message) {
+    writeToScreen("sent: " + message + '\n');
+    websocket.send(message);
+  }
+
+  //Function called to display message on screen
+  function writeToScreen(message) {
+    document.myform.outputtext.value += message
+    document.myform.outputtext.scrollTop = document.myform.outputtext.scrollHeight;
+  }
+
+
+  //Wrapper function for doSend()
+  function sendText() {
+    doSend(document.myform.inputtext.value);
+  }
+
+  //Function called to clear output text
+  function clearText() {
+    document.myform.outputtext.value = "";
+  }
+
+  //Function called to disconnect with Web-Sockets
+  function doDisconnect() {
+    websocket.close();
+  }
+
+  //Function to find image position on page
+  function findImagePosition(img) {
+    if (typeof (img.offsetParent) != "undefined") {
+      for (var x = 0, y = 0; img; img = img.offsetParent) {
+        x += img.offsetLeft;
+        y += img.offsetTop;
+      }
+      return [x, y];
+    }
+    else {
+      return [img.x, img.y];
+    }
+  }
+
+  //Function to get normalize coordinates of image
+  function GetImageCoordinates(evt) {
+    var x = 0;
+    var y = 0;
+    var imgPos;
+    imgPos = findImagePosition(img);
+    if (!evt) var evt = window.event;
+    if (evt.pageX || evt.pageY) {
+      x = evt.pageX;
+      y = evt.pageY;
+    }
+    else if (evt.clientX || evt.clientY) {
+      x = e.clientX + document.body.scrollLeft
+        + document.documentElement.scrollLeft;
+      y = e.clientY + document.body.scrollTop
+        + document.documentElement.scrollTop;
+    }
+
+    x -= imgPos[0];
+    if (x < 0) {
+      x = 0;
+    }
+    else if (x > img.width) {
+      x = img.width
+    }
+    x /= img.width;
+
+    y -= imgPos[1]
+    if (y < 0) {
+      y = 0;
+    }
+    else if (y > img.height) {
+      y = img.height
+    }
+    y /= img.height;
+    doSend("COORDINATES(x,y)=(" + x + "," + y+")")
+  }
+
+</script>
+
+
+<div id="output"></div>
+<form name="myform">
+  <div class="row">
+    <div class="column">
+      <img id="image" name="image" class="image" alt="">
+    </div>
+    <div class="column">
+      <textarea name="outputtext" rows="20" cols="50"></textarea>
+      <textarea name="inputtext" cols="50"></textarea>
+      <textarea name="url" cols="50"></textarea>
+      <br>
+      <input type="button" name=sendButton value="Send" onClick="sendText();">
+      <input type="button" name=clearButton value="Clear" onClick="clearText();">
+      <input type="button" name=disconnectButton value="Disconnect" onClick="doDisconnect();">
+      <input type="button" name=connectButton value="Connect" onClick="doConnect();">
+      </p>
+    </div>
+  </div>
+</form>
+<script>
+  //Getting reference of image

```

Comment:
This comment is a bit redundant - comments should normally tell why you do something, not what you do (that's clear from the code itself). E.g.:

// Attach mouse event to image to send the clicked coordinates to the back end
var img = document.getElementById("image");
img.onmousedown = GetImageCoordinates;